packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -0,0 +1,167 @@
+   *
+   * @param cameraId Camera identifier which indicates the device's camera for which to select a
+   *     {@link android.media.CamcorderProfile}.
+   * @param preset The {@link ResolutionPreset} for which is to be translated to a {@link
+   *     android.media.CamcorderProfile}.
+   * @return The best possible {@link android.media.CamcorderProfile} that matches the supplied
+   *     {@link ResolutionPreset}.
+   */
+  public static CamcorderProfile getBestAvailableCamcorderProfileForResolutionPreset(
+      int cameraId, ResolutionPreset preset) {
+
+    switch (preset) {
+        // All of these cases deliberately fall through to get the best available profile.
+      case max:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
+        }
+      case ultraHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_2160P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_2160P);
+        }
+      case veryHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_1080P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_1080P);
+        }
+      case high:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_720P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_720P);
+        }
+      case medium:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
+        }
+      case low:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_QVGA)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_QVGA);
+        }
+      default:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_LOW)) {
+          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
+        } else {
+          throw new IllegalArgumentException(
+              "No capture session available for current capture session.");
+        }
+    }
+  }
+
+  @VisibleForTesting
+  static Size computeBestPreviewSize(int cameraId, ResolutionPreset preset) {
+    if (preset.ordinal() > ResolutionPreset.high.ordinal()) {
+      preset = ResolutionPreset.high;
+    }
+
+    CamcorderProfile profile =
+        getBestAvailableCamcorderProfileForResolutionPreset(cameraId, preset);
+    return new Size(profile.videoFrameWidth, profile.videoFrameHeight);
+  }
+
+  private void configureResolution(ResolutionPreset resolutionPreset, int cameraId) {
+    recordingProfile =
+        getBestAvailableCamcorderProfileForResolutionPreset(cameraId, resolutionPreset);
+    captureSize = new Size(recordingProfile.videoFrameWidth, recordingProfile.videoFrameHeight);
+
+    previewSize = computeBestPreviewSize(cameraId, resolutionPreset);
+  }
```

Comment:
What's the order of the things in this class (per https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents)? I'm not seeing an obvious grouping.

Reply:
Fixed, there is no obvious (or hidden) grouping. I removed the empty line.

Reply:
The comment here wasn't about the empty line, but the ordering of methods in the class (see link)

Reply:
Apologies, I completely missed section 3.4.2 in the article and thought you referred to the "Formatting" section. I see what you mean now and update the order of methods so grouping is more logical.